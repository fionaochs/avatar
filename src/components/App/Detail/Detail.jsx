import React  from 'react';
import PropTypes from 'prop-types';


const Detail = ({ image, allies, enemies, affiliation, name }) =>  (
  <section>
    <h3>{name}</h3>
    <img src={image} />
    <p>Allies: {allies}</p>
    <p>Enemies: {enemies}</p>
    <p>Affiliation: {affiliation}</p>
  </section>
);


Detail.propTypes = {
  name: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  allies: PropTypes.string.isRequired,
  affiliation: PropTypes.string.isRequired,
  enemies: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
};

export default Detail;
