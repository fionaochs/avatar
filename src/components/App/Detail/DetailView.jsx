import React, { useState, useEffect }  from 'react';
import PropTypes from 'prop-types';
import { fetchCharacterDetail } from '../../../services/avatar';
import { useParams } from 'react-router-dom';


const Detail = () => {
  const [character, setCharacter] = useState([]);
  const params = useParams();

  useEffect(() => {
    fetchCharacterDetail(params.id)
      .then(res => setCharacter(res));
    console.log(params.id);
  }, []);
  return (
    <section>
      <h3>{character.name}</h3>
      <img src={character.image} />
      <p>Allies: {character.allies}</p>
      <p>Enemies: {character.enemies}</p>
      <p>Affiliation: {character.affiliation}</p>
    </section>
  );
};

Detail.propTypes = {
  name: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  allies: PropTypes.string.isRequired,
  affiliation: PropTypes.string.isRequired,
  enemies: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
};

export default Detail;
