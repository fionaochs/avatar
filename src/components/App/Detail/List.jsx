import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import Detail from './Detail';

const List = ({ characters }) => {
  const charactersList = characters.map(character => 
    <>
      <Link key={character.id} to={`characters/${character.id}`}>
        <li>
          <Detail {...character} />
        </li>
      </Link>
    </>
  );

  return (
    <ul>
      {charactersList}
    </ul>
  );
};


List.propTypes = {
  characters: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    allies: PropTypes.string.isRequired,
    affiliation: PropTypes.string.isRequired,
    enemies: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
  })).isRequired
};

export default List;
