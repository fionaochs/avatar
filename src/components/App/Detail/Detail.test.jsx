import React from 'react';
import { shallow } from 'enzyme';
import Detail from './Detail';

describe('Detail', () => {
  it('matches a detail snapshot', () => {
    const wrapper = shallow(<Detail allies='allies' name='Ang' enemies='enemies' affiliation='affiliation' image='image' />);
    expect(wrapper).toMatchSnapshot();
  });
});
