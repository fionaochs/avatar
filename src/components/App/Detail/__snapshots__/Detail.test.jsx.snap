// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail matches a detail snapshot 1`] = `
<section>
  <h3>
    Ang
  </h3>
  <img
    src="image"
  />
  <p>
    Allies: 
    allies
  </p>
  <p>
    Enemies: 
    enemies
  </p>
  <p>
    Affiliation: 
    affiliation
  </p>
</section>
`;
